mls-rs-uniffi/src/config.rs

```diff
@@ -37,7 +53,49 @@ pub type UniFFIConfig = client_builder::WithIdentityProvider<
     >,
 >;
 
-#[derive(Debug, Clone, uniffi::Record)]
+#[derive(Debug, Clone, uniffi::Object)]
 pub struct ClientConfig {
     pub group_state_storage: Arc<dyn GroupStateStorage>,
 }
+
+#[cfg(not(feature = "sqlite"))]
+#[uniffi::export]
+impl ClientConfig {
+    #[uniffi::constructor]
+    pub fn new(group_state_storage: Arc<dyn GroupStateStorage>) -> Self {
+        Self {
+            group_state_storage,
+        }
+    }
+}
+
+#[cfg(feature = "sqlite")]
+#[uniffi::export]
+impl ClientConfig {
+    #[uniffi::constructor]
+    pub fn new(group_state_storage: Arc<dyn GroupStateStorage>) -> Self {
+        Self {
+            group_state_storage,
+        }
+    }
```

Comment:
This function looks identical to the one in the block above when `sqlite` is off. Is there some `uniffi` limitation that forces us to duplicate it?

Reply:
Sorry for mixing unrelated changes by mistake. I reverted the commit.

Reply:
No problem. I was wondering why sqlite-related changes made it in this PR but I thought there must have been some reason I was not aware of :p